zerver/models/realms.py

```diff
@@ -1188,6 +1183,12 @@ def get_corresponding_policy_value_for_group_setting(
     # If the group setting is not set to one of the role based groups
     # that the previous enum setting allowed, then just return the
     # enum value corresponding to largest group.
+    if group_setting_name == "can_create_web_public_channel_group":
```

Comment:
I am not sure what is the best way to compute the largest group here.

One way can be that we can have a separate function which takes valid policy values as argument and then returns which is the largest among them.

Reply:
Don't we expect that the group should always be in `Realm.SYSTEM_GROUPS_ENUM_MAP`, given the restrictions on possible values? Not sure we need this case.

Reply:
Hmm, but it does have test coverage. I'm going to leave this one for you to investigate.

Reply:
Will handle while converting the next old setting.